.github/workflows/CI.yaml

```diff
@@ -0,0 +1,121 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - labeled
+    branches:
+      - main
+
+permissions:
+  id-token: write
+  contents: read
+  actions: read
+
+jobs:
```

Comment:
suggestion: can we at least split this file into multiple jobs?

As-is currently, if the tests were to fail, we would have to completely rebuild, stylelint, lint, build again, etc.

If we can at least split the tests into a separate job, and then publishing into a standalone job as well, it allows for only rerunning the job that likely failed.